src/commonMain/kotlin/fr/acinq/lightning/io/Peer.kt

```diff
@@ -1008,6 +1011,11 @@ class Peer(
                         val (feerate, fee) = watcher.client.computeSpliceCpfpFeerate(channel.commitments, targetFeerate, spliceWeight = weight, logger)
 
                         logger.info { "requesting splice-in using balance=${cmd.walletInputs.balance} feerate=$feerate fee=$fee" }
+                        if (channel.spliceStatus != SpliceStatus.None) {
+                            logger.info { "deferring splice, another splice is already in progress" }
+                            swapInCommands.send(SwapInCommand.UnlockWalletInputs(cmd.walletInputs.map { it.outPoint }.toSet()))
+                            return
+                        }
```

Comment:
This is a bit parallel to the main topic of the PR, but definitely needed too. I've seen conflicts between `UpdateFee` (from peer) and `SpliceInit` happen in the wild. In that case we stay stuck.

However I think we should do it at a different place if we want to really handle races properly, and instead define a `replyTo` here, that unlocks the inputs: https://github.com/ACINQ/lightning-kmp/blob/88b436644b731d194d7228c78a374d662b9c1557/src/commonMain/kotlin/fr/acinq/lightning/io/Peer.kt#L1019-L1024

Reply:
Good idea, it's much better to handle all failure cases with the `replyTo` field :+1: 
Done in https://github.com/ACINQ/lightning-kmp/pull/548/commits/9d6fdce2301514c022753466dbfa9106c972e34f